sirepo/pkcli/job_supervisor.py

```diff
@@ -205,6 +185,22 @@ def _sigterm(signum, frame):
     tornado.ioloop.IOLoop.current().add_callback_from_signal(_terminate)
 
 
+class _DataFileReq(tornado.web.RequestHandler):
+
+    async def put(self, path):
+        # should be exactly two levels
+        (d, f) = path.split('/')
+        assert sirepo.job.UNIQUE_KEY_RE.search(d), \
+            'invalid directory={}'.format(d)
+        d = sirepo.job.DATA_FILE_ROOT.join(d)
+        assert d.check(dir=True), \
+            'directory does not exist={}'.format(d)
+        # (tornado ensures no '..' and '.'), but a bit of sanity doesn't hurt
```

Comment:
i think we should either call validate_path or not, not do some other check.

Reply:
I don't think calling validate_path is appropriate because this code doesn't have anything to do with the simulation_db. It is just putting a file in a known location (outside of the simulation_db) where the flask server will look for it. Am I not understanding things properly?

Reply:
I see. I was confused.  Good checks then. :)